src/PdoConnection.php

```diff
@@ -248,7 +268,17 @@ public function rollBack()
      */
     public function commit()
     {
-        return $this->connect()->commit();
+        if (--$this->transactionCount === 0) {
+            if (!$this->rollbackNeeded) {
+                return $this->connect()->commit();
+            }
+
+            $this->rollbackNeeded = false;
+            $this->connect()->rollback();
+            return false;
+        }
+
+        return ($this->transactionCount >= 0 && !$this->rollbackNeeded);
```

Comment:
`return` `return` `return` :sob: 

imo, the only time it makes sense to return something other than `false` is if you `return $this->connect()->commit()`.